src/Swarm/Language/Syntax.hs

```diff
@@ -555,9 +555,12 @@ constInfo c = case c of
     command 0 long . doc "Deconstruct an old robot." $
       ["Salvaging a robot will give you its inventory, installed devices and log."]
   Say ->
-    command 1 short . doc "Emit a message." $ -- TODO: #513
-      [ "The message will be in a global log, which you can not currently view."
-      , "https://github.com/swarm-game/swarm/issues/513"
+    command 1 short . doc "Emit a message." $
+      [ "The messages can be viewed in the message window."
+      , "However, there is a limit to how distant messages you can hear."
```

Comment:
The problem with showing all messages in that place is that it might give the wrong impression that the robot was able to hear them. But it could have just moved to the radius of that message or conversely not see messages it would have been able to hear!

My plan was to be strict and just include those that the robot heard using the `listen` command. So just removing the duplicity in Creative mode and colouring those logs that are saying "I heard robot X say Y".

But that is not very user-friendly and in hindsight, I should show at least some messages.

A truly ~evil~ realistic thing to do would be to show the messages whose distance equals their age. Meaning that sound travels one cell each tick.
Litlle less ~evil~ realistic would be to have some interval like one real-world second to make this more user-friendly.
I am starting to like this idea - it can populate the message queue history and relativity is just cool.
Bonus points for checking that the message has a path to the robot that is not blocked! :rofl: 

Easier to implement would be to just show the messages in the radius and limit them to last real-world second - so 16 ticks.
But that is a bit annoying and does not make much sense.

Whichever way we choose, I feel it should be how `listen` finds messages, so that the players can use the message queue as a guide to how 

---

**TL;DR:** I propose I will implement one of the following options, which will form the logic of `listen`:
- show say messages of other robots in radius _and time limit of one real-world second_
- show say messages that would reach the robot travelling at one cell per tick
- show say messages that would reach the robot travelling at one cell per tick _with a tolerance of one real-world second_
- _show all messages_ as we are not pedantic relativity nerds and #577 is not implemented

What do you think @byorgey?

Reply:
Relativity is fun but doesn't seem relevant here, since even audio signals typically travel fast enough that the delay is not really noticeable within the range of the sensor.  When we get to #94 we should definitely come back to that idea.  

How about this: 
- Messages from other robots only show up in the Messages dialog *if you have a listening device installed*
- Then we just show say messages of other robots within the radius and a certain time limit.
    - Am I understanding correctly that by "a time limit of one real-world second" you mean that messages would flash on the Messages dialog for one second and then disappear again?
    - One second seems way too short to see and read any messages.
    - Why one real-world second?  That would be more complex to implement and I'm not sure what the benefit would be.  Why not just say "16 ticks" or whatever?

Reply:
> Am I understanding correctly that by "a time limit of one real-world second" you mean that messages would flash on the Messages dialog for one second and then disappear again?

Exactly, so it's not great. But maybe there is a better option that I forgot about:
- say will put the message to the global queue and to all robot logs of robots that:
  - are within the radius
  - have logger
  - have some listening device

Suggestion for "hearing aid" recipe? :smile: Maybe we can do #577 here too.

Reply:
Ah, right, if we put it in the logger then they will remember it so it doesn't matter if they move later. I like it!

For a hearing aid, maybe something like a copper pipe, some paper, a lodestone, and some wires?